snooty/postprocess.py

```diff
@@ -902,13 +902,15 @@ def find_toctree_nodes(
 
                     toctree_node = {
                         "title": title,
-                        "slug": slug,
+                        "slug": "/" if slug == "index" else slug,
                         "children": [],
                         "options": {"drawer": slug not in self.toc_landing_pages},
                     }
 
-                    new_ast = self.pages[slug_fileid].ast
-                    self.find_toctree_nodes(slug_fileid, new_ast, toctree_node)
+                    # Don't recurse on the index page
+                    if entry.slug != "/index":
+                        new_ast = self.pages[slug_fileid].ast
+                        self.find_toctree_nodes(slug_fileid, new_ast, toctree_node)
```

Comment:
Just thinking aloud, the idea here is to avoid recursing on a page we've already seen... could this be worth generalizing to handle that case? No problem if not!

Reply:
Contrary to my initial thought, there's basically no performance downside to this. I parsed and measured `docs`.

#### Before change
time: `1m4.247s`
memory usage: ~`831 MB`, brief spike to `1.0GB` near the end

#### After change
time: 1m4.687s
memory usage: ~`828 MB`, brief spike to `1.0GB` near the end